src/psyclone/domain/lfric/kern_call_arg_list.py

```diff
@@ -201,7 +203,7 @@ def field_vector(self, argvect, var_accesses=None):
             self.append(text)
         if var_accesses is not None:
             # We add the whole field-vector, not the individual accesses.
-            var_accesses.add_access(argvect.name, argvect.access,
+            var_accesses.add_access(Signature(argvect.name), argvect.access,
```

Comment:
We will be, since the variable access class stores the indices. I will decide later if we also need to store them in the signature (unlikely, tbh, e.g. ``a(1)%b`` and ``a(2)%b`` should have the same signature, but testing the indices will show that they do not overlap.

Reply:
Great.